presto-orc/src/main/java/com/facebook/presto/orc/metadata/ExceptionWrappingMetadataReader.java

```diff
@@ -114,7 +116,9 @@ public StripeFooter readStripeFooter(List<OrcType> types, InputStream inputStrea
 
     private OrcCorruptionException propagate(Throwable throwable, String message)
     {
-        propagateIfPossible(throwable, PrestoException.class);
+        propagateIfPossible(throwable, InvalidFunctionArgumentException.class);
+        propagateIfPossible(throwable, NotSupportedException.class);
```

Comment:
But checking the callsites, I think we need to catch NotSupportedException for getNextPage method in both OrcBatchPageSource and OrcSelectivePageSource. We also need to catch InvalidFunctionArgumentException for OrcBatchPageSource.getNextPage

Reply:
Actually i blindly replaced PrestoException here will all three. But looking deeper , will PrestoException ever be sent to ``` propagate ``` by its callers ? 
``` readBloomFilterIndexes(), readRowIndexes(), readStripeFooter() ``` etc ?

Reply:
Actually, is it even possible for `throwable` here to be InvalidFunctionArgumentException, NotSupportedException, etc?

Reply:
Its not possible. I think we should remove all ``` propagateIfPossible ``` ?

Reply:
SG, let's remove